Examples/Python/python/acts/examples/__init__.py

```diff
@@ -285,3 +286,39 @@ def _allmods(mod, base, found):
     if not quiet and donemods:
         print("dump_args for module functions:", ", ".join(donemods))
     return alldone
+
+
+class CustomLogLevel(Protocol):
+    def __call__(
+        self,
+        override: Optional[acts.logging.Level] = None,
+        min: acts.logging.Level = acts.logging.VERBOSE,
+        max: acts.logging.Level = acts.logging.FATAL,
+    ) -> acts.logging.Level:
+        ...
```

Comment:
Sorry, I'm not familiar with the use of a `Protocol` class. Is this just to keep the argument checker happy for the `defaultLogging` definition?
If so, do the `__call__` argument names need to match the `customLogLevel` argument names?

Reply:
this is also new to me. the idea was to have a type for `customLogLevel` which is returned by the `defaultLogging` function. my hope was also that vscode would pick that up and give you hints about `customLogLevel`. at least for me that did not work but I kept it to define the function

Reply:
I think you could use 
```python
Callable[[Optional[acts.logging.Level], acts.logging.Level, acts.logging.Level], acts.logging.Level]
```
as the type?

Reply:
my problem with this was that I could not express what the different parameters mean. but I guess from the vscode point of view it does not make a difference because the `Protocol` is not inspected right now anyway

Reply:
an alternative would be to make a class out of `defaultLogging` and make it callable. maybe IDEs could handle that better. on the other hand a class might look like we are wrapping the logger